libirods_rule_engine_plugin-elasticsearch.cpp

```diff
@@ -178,22 +190,44 @@ namespace
 				net::io_context ioc;
 
 				tcp::resolver resolver{ioc};
-				const auto results = resolver.resolve(result->host(), result->port());
+#if 0
+				tcp::resolver::results_type results;
 
-				if (use_tls) {
+				if (url_parse_result->has_port()) {
+					results = resolver.resolve(url_parse_result->host(), url_parse_result->port());
+				}
+				else if (url_parse_result->scheme_id() == urls::scheme::https) {
+					results = resolver.resolve(url_parse_result->host(), "443");
+				}
+				else if (url_parse_result->scheme_id() == urls::scheme::http) {
+					results = resolver.resolve(url_parse_result->host(), "80");
+				}
+#else
+				// If the admin does not define a port, ".port()" will return an empty string and
+				// communication with elasticsearch will not be possible.
+				const auto results = resolver.resolve(url_parse_result->host(), url_parse_result->port());
+#endif
```

Comment:
Elasticsearch only defines 9200 as the default port for API requests. I don't think we need to introduce any logic for defaulting to 80, 443, or 9200 based on the URL scheme. I think the admin should include the port in the config.

Reply:
agreed - use what they give us.  and we make the docs good.

Reply:
Done